add appveyor testing for windows

scottbelden opened the pull request:
Resolves https://github.com/tebeka/fastavro/issues/125
Resolves https://github.com/tebeka/fastavro/issues/126

@barrywhart I needed to change some of the `long`s to `long long` as it seems that for windows a `long` is only 4 bytes whereas on linux it is 8 bytes (which was causing some tests to fail on windows). I wasn't sure if this was the best approach and thought you might have more experience here.

barrywhart commented on tests/test_logical_types.py:
```diff
@@ -76,20 +78,25 @@ def test_logical_types():
-        'time-millis': 4,
-        'time-micros': 5,
+if os.name == 'nt' and sys.version_info[:2] == (3, 6):
+    @pytest.mark.skip(reason='Python Bug: https://bugs.python.org/issue29097')
```
You may want to look into using `skipif`.

https://docs.pytest.org/en/latest/skipping.html

barrywhart commented:
I gave it a quick look. It seems reasonable. So the "long long" type works okay on Linux, too?

If so, its fine. If not, then in regular C or C++, I would typically use a typedef to encapsulate types that may differ by platform. That way those details are isolated to one small part of the code. I'm not familiar with how that works in Cython, but it is probably possible. The idea would be to use types that specify the size, e.g. long64.

barrywhart commented:
More detail about the above... the `[ctypedef](http://cython.readthedocs.io/en/latest/src/userguide/language_basics.html)` keyword lets you define aliases for types. I think by combining `IF` statements (see the example in the same page linked above), we could set up aliases for 32 and 64-bit signed and unsigned integer types and use those throughout the `.pyx` files.

scottbelden commented:
Changed the test to clean up the conditional skip.

The `long long` works as far as I can tell. The original tests continue to pass. I added the relevant information to https://github.com/tebeka/fastavro/issues/128 as potential future work.

barrywhart commented:
About `long long`, it probably does work fine, since it sounds like `long long` and `long` are the same in Linux. The work I suggested could be helpful in the future if we encounter other OSes/compilers with incompatible types or behavior.

barrywhart commented on tests/test_logical_types.py:
```diff
@@ -77,6 +79,10 @@ def test_logical_types():
 def test_not_logical_ints():
+    # This test doesn't work on Windows with python 3.6
+    if os.name == 'nt' and sys.version_info[:2] == (3, 6):
+        pytest.skip('Python Bug: https://bugs.python.org/issue29097')
```
If you use `skipif`, you can pass the condition as part of the `skipif` function call. I think what you have works, this is more of a style thing.

barrywhart replied:
https://docs.pytest.org/en/latest/skipping.html#id1
